contents/tutorials/nextjs-ab-tests.md

```diff
@@ -0,0 +1,269 @@
+
+Once we create the experiment, we can start to implement it. Come back and launch it later.
+
+## Implementing our experiment
+
+When it comes to implementing our experiment and its associated feature flags, there are a few options we can choose:
+
+1. Bootstrap feature flags by using [`getInitialProps`](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props) in `_app.js`
+2. Server-side render feature flags in `index.js` using `posthog-node`.
+3. Client-side render feature flags in `index.js`.
+
+Because `getInitialProps` disables [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization) for the entire app (one of the main benefits of Next.js), we aren’t going to bootstrap feature flags (option 1). Instead, we use a combination of server-side and client-side rendering. This ensures flags' accuracy and speed while still leveraging the unique benefits of Next.js.
+
+### Client-side rendering feature flags
+
+To client-side render the experiment using feature flags, we can use the `PostHogProvider` we set up earlier. This ensures PostHog is loaded when we check the flags.
+
+In `index.js`, set up a state for the call to action button and a `useEffect()` to check the flag when the page loads.
+
+To check the flag, import `useFeatureFlagEnabled` from `posthog-js/react`, use it to call the flag name (`main-cta`), and change the state of the button text to "Learn more" if `true`.
+
+> **Note:** Use `posthog.feature_flags.override({'main-cta': 'test'})` to make sure it is working, but remove it when we go to release.
+
+```js
+// pages/index.js
+import Head from 'next/head'
+import { useEffect, useState } from 'react'
+import { usePostHog, useFeatureFlagEnabled } from 'posthog-js/react'
+
+export default function Home() {
+
+  const [ ctaState, setCtaState ] = useState('Click me')
+  const posthog = usePostHog()
+  posthog.feature_flags.override({'main-cta': 'test'})
+  const ctaVariant = useFeatureFlagVariantKey('main-cta')
+
+  useEffect(() => {
+    if (ctaVariant === 'test') {
+      setCtaState('Learn more')
+    }
+  }, [ctaVariant])
+
+  return (
+    <>
+      <Head>
+        <title>Next.js A/B tests</title>
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main>
+        <h1>Next.js A/B tests</h1>
+        <button id="main-cta">{ctaState}</button>
+      </main>
+    </>
+  )
+}
+```
+
+If you’ve set everything up correctly, this changes the call to action to "Learn more"
+
+![Learn call to action](../images/tutorials/nextjs-ab-tests/learn.png)
+
+With this you’re ready to launch your experiment, but make sure to remove your override.
+
+### Server-side rendering
+
+Notice when you refresh the call to action flickers between "Click me" and "Learn more." This is because it takes time for React and PostHog to load. 
```

Comment:
```suggestion
Notice when you refresh the call to action that it flickers between "Click me" and "Learn more." This is because it takes time for React and PostHog to load. 
```